### DISCUSSION TITLE
IntelliSense for config

### DISCUSSION THREAD

**Discussion by michaeloliverx** [pngwn/MDsveX#280] [Q&A] [answered]
Packages that ship with TypeScript typings can leverage IDE's intellisense with jsdoc type hints in JS files.

[Vite](https://vitejs.dev/config/#config-intellisense)
```js
/**
 * @type {import('vite').UserConfig}
 */
const config = {
  // ...
}

export default config
```

SvelteKit
```js
/**
 * @type {import('@sveltejs/kit').Config}
 */
const config = {
	// ...
};

export default config;
```

[Tailwind](https://github.com/tailwindlabs/tailwindcss/discussions/1077#discussioncomment-528222)
```js
/** @type {import("@types/tailwindcss/tailwind-config").TailwindConfig } */
```

Unfortunately I cannot get this working with MDsveX:
```js
/**
 * @type {import('mdsvex').MdsvexOptions}
 **/
const config = {
	extensions: ['.svx', '.md']
};

module.exports = config;
```

`config` is still `any`

I noticed when typing out the path for the others the types/ts files are listed:
<img width="522" alt="image" src="https://user-images.githubusercontent.com/55017335/125201564-a3a5e800-e267-11eb-96b5-e565410a703b.png">

But when doing the same for mdsvex no typescript files are shown:
<img width="462" alt="image" src="https://user-images.githubusercontent.com/55017335/125201612-cdf7a580-e267-11eb-8218-b96ba9ecfa87.png">

Perhaps its an issue how the types are being bundled? (I have no idea how the npm packaging works but just a guess)

--- 

Its not majorly important but it is _nice_ 😄

**Comment by pngwn** [comment]
Hmm. Interesting. 

I'd like to support this. Could you file an issue?

**Comment by pngwn** [comment, marked as answer]
This is now possible.

The best experience is using the `defineMDSvexConfig` (I'll need to change that name, very long):

```js
import { defineMDSveXConfig } from "mdsvex";

export default defineMDSveXConfig({
  extensions: [".svelte.md", ".md", ".svx"],

  smartypants: {
    dashes: "oldschool",
  },

  remarkPlugins: [],
  rehypePlugins: [],
});
```

Alternatively you can use the type comment syntax:

```js
/** @type {import('mdsvex').MdsvexOptions} */
export default {
  ...
}
```

**Comment by pngwn** [comment]
This is now possible.

The best experience is using the `defineMDSvexConfig` (I'll need to change that name, very long):

```js
import { defineMDSveXConfig } from "mdsvex";

export default defineMDSveXConfig({
  extensions: [".svelte.md", ".md", ".svx"],

  smartypants: {
    dashes: "oldschool",
  },

  remarkPlugins: [],
  rehypePlugins: [],
});
```

Alternatively you can use the type comment syntax:

```js
/** @type {import('mdsvex').MdsvexOptions} */
export default {
  ...
}
```
